src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/EventsShouldNotHaveBeforeOrAfterPrefix.cs

```diff
@@ -25,31 +22,31 @@ public sealed class EventsShouldNotHaveBeforeOrAfterPrefix : DiagnosticAnalyzer
             s_localizableTitle,
             s_localizableMessage,
             DiagnosticCategory.Naming,
-            RuleLevel.IdeSuggestion,
+            RuleLevel.Disabled,
             description: s_localizableDescription,
             isPortedFxCopRule: true,
             isDataflowRule: false);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
+        private static readonly ImmutableHashSet<string> s_invalidPrefixes = ImmutableHashSet.Create("After", "Before");
+
         public override void Initialize(AnalysisContext analysisContext)
         {
             analysisContext.EnableConcurrentExecution();
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
             analysisContext.RegisterSymbolAction(context =>
             {
-                if (IsViolatingPrefix(context.Symbol.Name, AfterKeyword) ||
-                    IsViolatingPrefix(context.Symbol.Name, BeforeKeyword))
+                var wordParse = new WordParser(context.Symbol.Name, WordParserOptions.SplitCompoundWords);
```

Comment:
We don't want to do this unless we know it is likely that the symbol name starts with "Before" or "After". Can we have a cheaper bail out check upfront?

Reply:
Sure! Do you think the `WordParser` is valuable? I could revert to my previous implementation if needed.

Reply:
It seems fine to use `WordParser`. Just avoid the perf hit of invoking it for every symbol.